apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/DatasetsResourceTest.java

```diff
@@ -3697,6 +3698,64 @@ static Stream<Arguments> find__whenFilterInvalidOperatorForFieldType__thenReturn
         }
     }
 
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    //TODO: Remove this test class after migration to the new dataset item format
+    class TestNoMigratedDatasetItemRetrieval {
+
+        private DatasetItem datasetItem;
+
+        @BeforeEach
+        void setUp() {
+
+            var datasetId = createAndAssert(factory.manufacturePojo(Dataset.class));
+            var clickhouseConnectionFactory = ClickHouseContainerUtils.newDatabaseAnalyticsFactory(
+                    CLICKHOUSE, DATABASE_NAME).build();
+
+            var datasetItem = factory.manufacturePojo(DatasetItem.class);
+
+            Mono.from(clickhouseConnectionFactory.create())
+                    .flatMap(connection -> Mono.from(connection.createStatement("""
+                    INSERT INTO %s.%s (
```

Comment:
Inserting directly from a resource test isn't great for maintainability (it'd be from a DAO test). But as the TODO mentions, this is temporary. Also, there's no other possible way of doing it. So this is fine.